src/bootstrap/config.rs

```diff
@@ -405,7 +405,7 @@ impl Config {
         config.incremental = flags.incremental;
         config.dry_run = flags.dry_run;
         config.keep_stage = flags.keep_stage;
-        if let Some(value) = flags.warnings {
+        if let Some(value) = flags.deny_warnings {
             config.deny_warnings = value;
         }
```

Comment:
So argument parsing here is some kind of two-stage process? `&[String]` -> `Flags` -> `Config`? And this is specifically for what gets passed on the cmdline?

Reply:
Yep, that's correct -- configuration loading is a bit non-ideal.